xray/segment.go

```diff
@@ -54,6 +54,25 @@ func BeginFacadeSegment(ctx context.Context, name string, h *header.Header) (con
 	return context.WithValue(ctx, ContextKey, seg), seg
 }
 
+// Begin DummySegment creates a segment in the case of no sampling to reduce memory footprint
+func BeginDummySegment(ctx context.Context, name string) (context.Context, *Segment) {
+	dummySeg := &Segment{parent: nil}
+	dummySeg.ParentSegment = dummySeg
+	logger.Debugf("Beginning segment named %s", name)
+
+	cfg := GetRecorder(ctx)
+	dummySeg.assignConfiguration(cfg)
```

Comment:
Is it necessary to pass the recorder in configuration for a DummySegment? The `assignConfiguration` method does a bunch of comparisons which should be avoided if we don't really need the configuration.

Reply:
right.